net-p2p/mldonkey/mldonkey-3.1.7.ebuild

```diff
@@ -0,0 +1,181 @@
+
+RDEPEND="dev-lang/perl
+	dev-ml/camlp4:=
+	gd? ( media-libs/gd[truetype] )
+	gtk? (
+		gnome-base/librsvg
+		dev-ml/lablgtk[svg]
+	)
+	guionly? (
+		gnome-base/librsvg
+		dev-ml/lablgtk
+	)
+	magic? ( sys-apps/file )
+	upnp? (
+		net-libs/libnatpmp
+		net-libs/miniupnpc:=
+	)
+	acct-user/p2p
+"
+
+DEPEND="${RDEPEND}
+	<dev-lang/ocaml-4.10[ocamlopt?]
+	bittorrent? (
+		|| (
+			<dev-lang/ocaml-4.06
+			(
+				>=dev-lang/ocaml-4.06
+				dev-ml/num
+			)
+		)
+	)"
+
+RESTRICT="!ocamlopt? ( strip )"
+
+S="${WORKDIR}/${P}-2"
+
+pkg_setup() {
+	if use gtk; then
+		echo ""
+		einfo "If the compile with gui fails, and you have updated Ocaml"
+		einfo "recently, you may have forgotten that you need to run"
+		einfo "/usr/portage/dev-lang/ocaml/files/ocaml-rebuild.sh"
+		einfo "to learn which ebuilds you need to recompile"
+		einfo "each time you update Ocaml to a different version"
+		einfo "see the Ocaml ebuild for details"
+		echo ""
+	fi
+
+	# dev-lang/ocaml creates its own objects but calls gcc for linking, which will
+	# results in relocations if gcc wants to create a PIE executable
+	if gcc-specs-pie ; then
+		append-ldflags -nopie
+		ewarn "Ocaml generates its own native asm, you're using a PIE compiler"
+		ewarn "We have appended -nopie to ocaml build options"
+		ewarn "because linking an executable with pie while the objects are not pic will not work"
+	fi
+}
+
+src_prepare() {
+	cd config || die
+	eautoconf
+	cd .. || die
+	if ! use ocamlopt; then
+		sed -i -e "s/ocamlopt/idontwantocamlopt/g" "${S}/config/configure" || die "failed to disable ocamlopt"
+	fi
+
+	default
+}
+
+src_configure() {
+	myconf=""
+
+	if use gtk || use guionly; then
+		myconf+="--enable-gui=newgui2"
+	else
+		myconf+="--disable-gui"
+	fi
+
+	if use guionly; then
+		myconf+="--disable-multinet --disable-donkey"
+	fi
+
+	local my_extra_libs
+	if use gd; then
+		my_extra_libs="-lpng"
+	fi
+
+	econf LIBS="${my_extra_libs}"\
+		--sysconfdir=/etc/mldonkey \
+		--sharedstatedir=/var/mldonkey \
+		--localstatedir=/var/mldonkey \
+		--enable-checks \
+		--disable-batch \
+		$(use_enable bittorrent) \
+		$(use_enable fasttrack) \
+		$(use_enable gnutella) \
+		$(use_enable gnutella gnutella2) \
+		$(use_enable gd) \
+		$(use_enable magic) \
+		$(use_enable upnp upnp-natpmp) \
+		--disable-force-upnp-natpmp \
+		${myconf}
+}
+
+src_compile() {
+	export OCAMLRUNPARAM="l=256M"
+	emake -j1 # Upstream bug #48
+
+	if ! use guionly; then
+		emake utils
+	fi
+}
+
+src_install() {
+	local myext=""
+	use ocamlopt || myext=".byte"
+	if ! use guionly; then
+		for i in mlnet mld_hash get_range copysources subconv; do
+			newbin "${i}${myext}" "${i}"
+		done
+		use bittorrent && newbin "make_torrent${myext}" make_torrent
+
+		newconfd "${FILESDIR}/mldonkey.confd-2.8" mldonkey
+		fperms 600 /etc/conf.d/mldonkey
+		newinitd "${FILESDIR}/mldonkey.initd" mldonkey
+	fi
+
+	if use gtk; then
+		for i in mlgui mlguistarter; do
```

Comment:
"local i=" above this

Reply:
As far as I know, variables can be declared as local at the function scope only, so does it make sense to declare the same variable [twice](https://github.com/gentoo/gentoo/pull/16648#discussion_r483980009) in the same function?
Instead, I declared `i` as local at the beginning of the function, I hope that make sense.

Reply:
Sure!